app/scripts/lib/metrics.js

```diff
@@ -484,10 +484,16 @@ define(function (require, exports, module) {
 
     logFlowEvent (eventName, viewName) {
       this.logEvent(marshallFlowEvent(eventName, viewName));
+      // We need to use server-side timestamps on flow events,
+      // so flush them to the server immediately.
+      this.flush();
```

Comment:
Ah, my bad, I didn't realise your objection was as strong as that in the other PR, sorry. I'll close this and come up with something else!

> ...where you say "clients are sending garbage flow" is it because the timestamps are totally wrong?

Yep, we received some flow events from the distant future!